src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -204,7 +204,12 @@ private void SetNLogElementSettings(ILoggingConfigurationElement nlogConfig)
             {
                 if (!string.IsNullOrEmpty(configItem.Key))
                 {
-                    dict[configItem.Key.Trim()] = configItem.Value;
+                    string key = configItem.Key.Trim();
+                    if (dict.ContainsKey(key))
+                    {
```

Comment:
This is something else than the refactor, isn't?

Reply:
Are there more of these changes in this PR?

Reply:
Yes some debug-noise to help diagnose Xml-config issues.